DocumentedExamples/Submitting_analysis_jobs_to_gadi.ipynb

```diff
@@ -0,0 +1,275 @@
+   "id": "beee3fe0-8aee-43fd-bcd8-52f1039639d9",
+   "metadata": {},
+   "source": [
+    "# Submitting analysis jobs to gadi\n",
+    "\n",
+    "This notebook goes through an example of how to convert your jupyter notebook code into a PBS job you can submit to Gadi, NCI.\n",
+    "\n",
+    "Sometimes, once you have your analysis working in a jupyter notebook and you are using a lot of data, it can be helpful to submit the job directly to Gadi. This uses the PBS queue (see NCI website for more details, https://opus.nci.org.au/display/Help/4.+PBS+Jobs). It means the job is not interactive, but it also means you can submit many jobs at the same time if say you want to break up your analysis and run one year/month at a time.\n",
+    "\n",
+    "One way to set up this analysis is to convert your `.ipynb` notebooks into `.py` files, and then run on gadi with a wrapping `.sh` bash script. Functions and packages that work in your `.ipynb` notebooks, such as `cosima_cookbook` should also work on gadi, however `dask` requires some special treatment. See the CLEX CMS team's article on this for more information  (https://coecms-training.github.io/parallel/dask-batch.html).\n",
+    "\n",
+    "This notebook shows how to \n",
+    "1. set up your `.py` script to use `dask` stably\n",
+    "2. set up your PBS script\n",
+    "3. Optional: add an environment variable for easy submission of different months/years/variables"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "743d04d5-e4ca-4235-8be5-94bc6e59e8cf",
+   "metadata": {},
+   "source": [
+    "## Step 1: `.py` script\n",
+    "\n",
+    "``` \n",
+    "\"\"\"\n",
+    "Title - e.g. Computing EKE for one year\n",
+    "\"\"\"\n",
+    "\n",
+    "# Load modules as normal\n",
+    "\n",
+    "import matplotlib.pyplot as plt\n",
+    "import xarray as xr\n",
+    "from xgcm import Grid\n",
+    "import numpy as np\n",
+    "import pandas as pd\n",
+    "import cftime\n",
+    "from xhistogram.xarray import histogram\n",
+    "import cosima_cookbook as cc\n",
+    "\n",
+    "import IPython.display\n",
+    "import cmocean as cm\n",
+    "import cartopy.crs as ccrs\n",
+    "import cartopy.feature as cft\n",
+    "\n",
+    "import sys, os\n",
+    "import warnings\n",
+    "warnings.simplefilter(\"ignore\")\n",
+    "from dask.distributed import Client\n",
+    "\n",
+    "import climtas.nci\n",
+    "\n",
+    "if __name__ == '__main__':\n",
+    "\n",
+    "    climtas.nci.GadiClient()\n",
+    "    session = cc.database.create_session()\n",
+    "\n",
+    "    ## Your jupyter notebook code goes here inside the `main` loop\n",
+    "```\n",
+    "\n",
+    "Add your python code from jupyter as normal inside the main loop (be careful of indentation!). Note that at the end of the PBS script if you don't save anything out, everything will get deleted. So be sure to save any figures, data you need. For example, you could save out the variable EKE as a netcdf using the following code.\n",
```

Comment:
What does "as normal" mean to say? Perhaps delete "as normal"?

```suggestion
    "Add your python code from jupyter inside the main loop. **Be very careful of the indentation!**. Also note that at the end of the PBS script you need to explicitly save things, otherwise all the results of your analyses will get deleted. So, be sure to save any figures, data you need! For example, you could save out the variable EKE as a `netcdf` using the following code.\n",
```